aurora/interface/sample/from_id.py

```diff
@@ -53,24 +59,343 @@ class SampleFromId(ipw.VBox):
+        "margin": "0 auto 10px auto",
+        "height": "150px",
+        "max_height": "150px",
+        "overflow": "auto",
+        "align_items": "center",
+    }
+
     def __init__(
         self,
         experiment_model: BatteryExperimentModel,
         validate_callback_f: Callable,
     ) -> None:
+        """Constructor for the sample-selection-by-id widgets
+        container.
+
+        Parameters
+        ----------
+        `experiment_model` : `BatteryExperimentModel`
+            The current battery experiment model instance.
+        `validate_callback_f` : `Callable`
+            A callback function for the validation button.
+
+        Raises
+        ------
+        `TypeError`
+            If the callback function is not callable.
+        """
+
+        if not callable(validate_callback_f):
+            raise TypeError(
+                "validate_callback_f should be a callable function")
 
         self.experiment_model = experiment_model
 
-        # initialize widgets
-        self.w_header_label = ipw.HTML(value="<h2>Battery Selection</h2>")
+        filters_container = self._build_filter_container()
+
+        selection_container = self._build_selection_container()
+
+        self.w_validate = ipw.Button(
+            description="Validate",
+            button_style='success',
+            tooltip="Validate the selected sample",
+            icon='check',
+            disabled=True,
+            style=self.VALIDATE_BUTTON_STYLE,
+            layout=self.VALIDATE_BUTTON_LAYOUT,
+        )
+
+        super().__init__(
+            layout={},
+            children=[
+                filters_container,
+                selection_container,
+                self.w_validate,
+            ],
+        )
+
+        self._initialize_filters()
+
+        self._initialize_selectors()
+
+        self._set_event_listeners(validate_callback_f)
+
+    @property
+    def current_specs(self) -> dict:
+        """A dictionary representing the current selected specs used
+        to filter the available samples.
+
+        Returns
+        -------
+        `dict`
+            A dictionary of selected specs.
+        """
+        return {
+            'manufacturer': self.w_specs_manufacturer.value,
+            'composition.description': self.w_specs_composition.value,
+            'capacity.nominal': self.w_specs_capacity.value,
+            'form_factor': self.w_specs_form_factor.value,
+            # 'metadata.creation_datetime': self.w_specs_metadata_creation_date.value,
+        }
+
+    @property
+    def selected_sample_ids(self) -> Tuple[str, ...]:
+        return self.w_sample_list.value
+
+    @property
+    @remove_empties_from_dict_decorator
+    def selected_sample_dict(self) -> List[dict]:
+        """A list of sanitized sample dictionaries. Key/null value
+        pairs are discarded.
+
+        Returns
+        -------
+        `List[dict]`
+            A list of non-null sample dictionaries.
+        """
```

Comment:
A bit pedantic, but these should be `selected_samples_ids` and `selected_samples_dict`? Also, I'm surprised we are not getting any complains that `selected_samples_ids` does not have a docstring (in aiida core we would get complaints, but maybe there is a pre-commit we didn't add yet).

Reply:
I try to avoid repeated use of plural, so `samples` or `sample_ids`, but not `samples_ids`. But `samples_dict` is okay. I'll update!

Reply:
I've been avoiding docstrings on simple `property` methods, but sure, I can add them. Best to be consistent.

Reply:
> I try to avoid repeated use of plural, so `samples` or `sample_ids`, but not `samples_ids`. But `samples_dict` is okay. I'll update!

Mmm but why do you need to set a convention for this? Since all combinations refer to different possibilities. So I would use `sampe_id` for the id of a single sample, `sample_ids` when a single sample has many ids, `sampes_id` when you have a single id for a set of samples, and `samples_ids` for our case here. Note: I'm not saying we need to distinguish between all these cases for our use (we don't have multiple IDs for a single sample), only that these situations can exist so I would just use the best description for our case.

Anyway, is not a big deal so you can just merge away if you prefer to keep it like this.

Reply:
It's not MY convention, but rather a convention of the English language. For example, you wouldn't say "Which dogs names do you prefer?", but "Which dog names...?"

Anyhow, I think I'll just merge as is. The names of these two methods wasn't changed in this PR. We can address it in a later one if necessary.

Reply:
Very true :) I'll address the naming convention the next time I touch that file, likely when I return to implement the group filter.